src/prefect/context.py

```diff
@@ -290,7 +290,7 @@ async def get_or_create(cls) -> AsyncGenerator[Self, None]:
         if ctx:
             yield ctx
         else:
-            with cls() as ctx:
+            async with cls() as ctx:
```

Comment:
Without calling `__aenter__` here, the `AsyncClientContext.client` context is never opened as well. This means that `PrefectClient._loop` is never set, which means `get_client()` will always return a new client instead of the existing client.

https://github.com/PrefectHQ/prefect/blob/main/src/prefect/client/orchestration.py#L206